### ISSUE TITLE
Irrational Wanderer behaviour after the Unfettered invasion

### ISSUE THREAD

**Issue by tatami4** [endless-sky/endless-sky#5329]
**Describe the bug**
Wanderer ships fly to systems controlled by Unfettered after the Unfettered take some of Wanderer systems.

**Expected behavior**
Since Wanderers are peaceful, there is no reason for them to fly to Unfettered systems just to pointlessly fight (and die).

**Comment by tehhowch** [comment]
This is true of all ships for all governments: it is not very likely that a fleet will select a destination that is hostile to itself, but it is not improbable. Nor should it be (e.g. unfettered vs. traditional Hai).

We would need to implement a manner of customizing this chance based on characteristics of the given group of ships and/or their government / personality.

**Comment by Terin** [comment]
Such a feature, while probably not simple to implement, would come in handy for things like #5291 or the Unfettered not jumping to other systems outside of Wah Ki, on top of this. Might be worth considering.

**Comment by Zatishko** [comment]
There is already a work around between human and hai space with the Wormhole Alpha government, I think involving the wormhole. I don't especially understand how it works as for travel and one of those fleets have jump drives and the unfettered do. Been trying to work that out for a while now.

 How to put such a system in with no intervening stars would be problematical.

**Comment by arkhne** [comment]
While I see no simple solution other than "pretend it doesn't happen, since it isn't really a big deal", I think it would be nice if fleets checked the government of adjacent systems before jumping (and aborting the Jump if the tag found is incompatible). 

This way, you could set Unfettered to target Wanderer, Unfettered, and Hai systems, which would keep their behaviour "as expected", but also prevent the unlikely scenario of JD enabled Unfettered from hopping to Sestor space. While having Wanderers set to "Wanderer Only", which would keep them out of Unfettered-conquered Systems. This could also be used to Limit Hai and Human fleets from not entering Unfettered Space, etc. Such a system would be useful, but ultimately may turn out to be more effort than it is worth.

**Comment by Amazinite** [comment]
Perhaps this could be made a fleet/npc personality. 
* no new personality: retain existing behavior where they just jump to any adjacent system.
* `defensive`: don't jump to a hostile system if a friendly system can be jumped to. (Or perhaps just never jump to a hostile system, whichever we prefer.)
* `offensive`: if a hostile system can be jumped to, jump to it. (Or perhaps just weight jumping to a hostile system more heavily than a friendly one.)

**Comment by tehhowch** [comment]
The personality tokens should be concrete enough that seeing them attached to a ship definition is not confusing, so I don't think `defensive` and `offensive` are that great, as they could be construed to have an effect on the ship's in-combat behavior.

`warring` might be a good option for the "prioritizes hostile action" case. I can see that applying to jumps, and also increasing the range/strength difference needed to make the ship avoid combat with a given enemy.

**Comment by Fzzr** [comment]
In the same lore concern but a different behavior, see https://www.reddit.com/r/endlesssky/comments/10m6go3/so_this_exploit_still_works_spoiler/j6glsrn/

> Wanderer ships appear in the systems that they gave up, well after the invasion is over, and attack the invaders.
That doesn't really fit with their nature, nor the concept that they truly retreated and gave up those systems.

**Comment by Amazinite** [comment]
This can now be fixed by #6489.
